cou/utils/app_utils.py

```diff
@@ -50,6 +50,31 @@ async def upgrade_packages(
         await model.run_on_unit(unit_name=unit, command=command, timeout=600)
 
 
+async def get_instance_count(unit: str, model: COUModel) -> int:
+    """Get instance count on a nova-compute unit.
+
+    :param unit: Name of the unit where the action runs on.
+    :type unit: str
+    :param model: COUModel object
+    :type model: COUModel
+    :return: Instance count of the unit
+    :rtype: int
+    :raises ValueError: When the action result is not valid.
+    """
+    action_name = "instance-count"
+    action = await model.run_action(unit_name=unit, action_name=action_name)
+
+    if (
+        instance_count := action.results.get("instance-count").strip()
```

Comment:
not-blocking: What is the benefit of walrus operator here? IMO it is making it only less readable.

Reply:
After second thought, this is actually blocking comment, because if `"instance-count"` does not exist in action results, the get function will return `None`. Like this function can end up in `AttributeError`.

Reply:
I chose walrus operator because this variable is not used outside this conditional block. 

I agree with the issue you described and will add a default value for "get()" in this case.